content/blog/2018-07-17-pkginspector.md

```diff
@@ -0,0 +1,112 @@
+## 4                plasma (n, alpha = 1, begin = 0, end = 1, direction = 1) 
+## 5 viridis (n, alpha = 1, begin = 0, end = 1, direction = 1, option = "D") 
+## 6      viridisMap (n = 256, alpha = 1, begin = 0, end = 1, direction = 1, 
+##   calls called_by dependents
+## 1     1         0          0
+## 2     1         0          0
+## 3     1         0          0
+## 4     1         0          0
+## 5     0         4          4
+## 6     0         0          0
+```
+
+
+
+The example shows that the number of functions called by `cividis()`, `inferno()`, `magma()` and `plasma()` is 1, 1, 1 and 1, and that these functions are called by 0, 0, 0 and 0 functions. `viridis()`, in contrast, calls 0 functions but is called by 4 functions. In this case, the number of `dependents` is 4. Dependents are counted recursively and include any functions in the calling chain. For example, if A calls B and B calls C, we would say that C is called by 1 (B) but has 2 dependents (A & B). `rev_fn_summary()` also provides information about function parameters in the `f_args` column.
+
+*What's not working:* We know that we miss function calls if they are passed as parameters to `purrr::map()` and `do.call()` functions. There may be other systematic misses as well.
+
+### Visualization
+
+`vis_package()` helps you visualize the network of functions' dependencies ([interactive example](http://rpubs.com/jtr13/vis_package)).
+
+
+```r
+vis_package(path, physics = FALSE)
+```
+
+![](img/blog-images/2018-07-17-pkginspector/viridisLite.png)
+
+### Argument default usage
+
+`rev_args()` identifies all the functions' arguments used in a given package. It returns a dataframe which main column, `detault_consistent` indicates whether or not the default value of the argument is consistent across the functions that use it. This helps to evaluate the complexity of the package and to identify potential sources of confusion, for example, if the meaning or default value of the same argument varies across functions.
+
+
+```r
+rev_args(path)$arg_df
+```
+
+```
+##    arg_name n_functions default_consistent default_consistent_percent
+## 1         n           6              FALSE                   83.33333
+## 2     alpha           6               TRUE                  100.00000
+## 3     begin           6               TRUE                  100.00000
+## 4       end           6               TRUE                  100.00000
+## 5 direction           6               TRUE                  100.00000
+## 6    option           2               TRUE                  100.00000
+```
+
+The example shows that the parameter `n` is used inconsistently. [The documentation](https://github.com/sjmgarnier/viridisLite/blob/master/R/viridis.R) reveals that the default value of `n` is 256 in one function but missing in all others. This flags a potential issue that deserves further investigation. In this case, the odd function out--`viridisMap()`--has a clear use case that is different from the others.
```

Comment:
say "The package documentation" or "The viridis documentation" or something to be clear that you're not referring to docs for `pkginspector`

Reply:
> [The documentation](https://github.com/sjmgarnier/viridisLite/blob/master/R/viridis.R) reveals that the default value of `n` is 256 in one function but missing in all others. 

to

The documentation for  [`viridisLite::viridis()`](https://github.com/sjmgarnier/viridisLite/blob/master/R/viridis.R) reveals that the default value of `n` is 256 in one function but no default in all others.